### PR TITLE
build: update Kubernetes dependencies to 1.21.0

### PR DESCRIPTION
**Pull request by pohly**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

We would like to use 1.21.0 because of the CSIStorageCapacity beta API.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -29,72 +29,71 @@ require (
...
 	k8s.io/client-go v1.5.2
-	k8s.io/component-base v0.21.0-beta.1
-	k8s.io/component-helpers v0.21.0-beta.1
-	k8s.io/csi-translation-lib v0.21.0-beta.1
+	k8s.io/component-base v0.21.0-rc.0
+	k8s.io/component-helpers v0.21.0-rc.0
+	k8s.io/csi-translation-lib v0.21.0-rc.0
 	k8s.io/klog/v2 v2.8.0
 	k8s.io/kube-openapi v0.0.0-20210305164622-f622666832c1 // indirect
 	k8s.io/kubernetes v1.20.0
```
Comment (chrishenzie) [inline go.mod:41]: Is this also worth bumping?

**Comment by chrishenzie** [comment]
Do you want this PR merged as-is? We could re-use this to bump to 1.21 once it's released.

**Comment by chrishenzie** [comment]
Bumping, [v1.21.0](https://github.com/kubernetes/kubernetes/releases/tag/v1.21.0)  just released.

**Comment by pohly** [comment]
I tried updating this PR, but the mirrored repos have not been updated yet:
```
go: k8s.io/api@v0.21.0: reading k8s.io/api/go.mod at revision v0.21.0: unknown revision v0.21.0
```

**Comment by pohly** [comment]
We can either take it as-is or wait.

**Comment by chrishenzie** [comment]
If it helps unblock you we can proceed and then open a PR to update it again afterwards.

**Comment by pohly** [comment]
I'm fine with both the current beta1 code (no need to merge anything), merging this PR as-is (slightly more up-to-date) or waiting until the mirrors catch up.

**Comment by pohly** [comment]
More important is to get PRs #583 and #586 included.

**Comment by chrishenzie** [comment]
Ok, let's wait until v1.20 mirrors are ready.

**Comment by chrishenzie** [comment]
Mirrors are ready now, updating to 1.21.0 should work.

File: go.mod (hunk 1 above)
Comment (pohly) [reply go.mod:41]: Yes, done.

**Comment by pohly** [comment]
This PR is now updating to 1.21.0.

File: go.mod [hunk 2]
```diff
@@ -29,72 +29,71 @@ require (
 	google.golang.org/grpc v1.36.0
 	google.golang.org/protobuf v1.26.0
 	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b // indirect
-	k8s.io/api v0.21.0-beta.1
-	k8s.io/apimachinery v0.21.0-beta.1
-	k8s.io/apiserver v0.21.0-beta.1
+	k8s.io/api v0.21.0
+	k8s.io/apimachinery v0.21.0
+	k8s.io/apiserver v0.21.0
 	k8s.io/client-go v1.5.2
```
Comment (msau42) [inline go.mod:35]: Should we update this to v0.21.0? v1.5.0 is super ancient...

Comment (pohly) [reply go.mod]: Further down we have
```
replace k8s.io/client-go => k8s.io/client-go v0.21.0
```
i.e. in practice we are already using 0.21.0 as part of this PR.

I've fixed this line to reflect that.

**Pull request merged**